File: data/src/main/java/de/holtmeyer/niklas/spotify/data/entity/dto/UserProfile.java
package de.holtmeyer.niklas.spotify.data.entity.dto;

import de.holtmeyer.niklas.spotify.data.entity.dto.common.ExternalUrls;
import de.holtmeyer.niklas.spotify.data.entity.dto.common.HasHref;
import de.holtmeyer.niklas.spotify.data.entity.dto.common.HasHrefWithID;
import de.holtmeyer.niklas.spotify.data.entity.dto.common.Image;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper=true)
public class UserProfile extends HasHrefWithID {
    String country;
    String display_name;
    String email;
    ExplicitContent explicit_content;
    ExternalUrls external_urls;
    Followers followers;
    Image[] images;
    String product;
}

File: frontend/src/app/interface/UserProfile.ts
import { ExplicitContent } from './ExplicitContent';
import { ExternalUrls } from './ExternalUrls';
import { Followers } from './Followers';
import { Image } from './Image';
export interface UserProfile{
  country: String;
  display_name: String;
  email: String;
  explicit_content: ExplicitContent;
  external_urls: ExternalUrls;
  followers: Followers;
  href: String;
  id: String;
  images: Image[];
  product: String;
  type: String;
  uri: String;
}

